code-challenges.test.js

```diff
@@ -10,32 +14,264 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
+describe("aboutMe", () => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+    const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
+    const peopleOutput = ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+    expect(aboutMe(people)).toEqual(peopleOutput)
+  })
+})
+
+    // Jest Test RTN:
+    // ReferenceError: aboutMe is not defined
+    // Test Suites: 1 failed, 1 total
+    // Tests:       1 failed, 1 total
+
+
+const people = [
+  { name: "ford prefect", occupation: "a hitchhiker" },
+  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+  { name: "arthur dent", occupation: "a radio employee" }]
+
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
 // b) Create the function that makes the test pass.
+// Pseudocode:
+        //Paramenter will be an array of objects
+        //Argument: const people, an array of objects
+        //Create a function named aboutMe
+        //Get Destructive on the array of objects and destructure it twice to seperate each object key value into two seperate arrays.  Then use .map on the object value on the key name: to pull out the names then .split.map(value items).charAt(0).toUpperCase, add capital letter to begining of each value item.  Also need to seperate the second key item using [...new Set()] constructor. Seperate all destructured items and then create the sentance with new variables to return.  I think I just recreated the same array object for the name destructering, and I dont think I needed the Object constructor but it worked and I could .map.map so I kept it, not really understanding it though since it just made an object from an object array which i know I cant .map an object but the (array) was still there.
+        //Return the array with string interpolation, it will have a sentence about each person with their name capitalized.
+
+const aboutMe = (array) => {
+        let work = [...new Set(array.map(item => item.occupation))];
+        let name = (Object.values(array).map((value)=>value.name.split(" ").map(item=> item.charAt(0).toUpperCase() + item.substr(1))))
+            let one = name[0].join(' ')
+            let one1 = work[0]
+            let two = name[1].join(' ')
+            let two2 = work[1]
+            let three = name[2].join(' ')
+            let three3 = work[2]
+        return [`${one} is ${one1}.`, `${two} is ${two2}.`, `${three} is ${three3}.`]
+        }
+ console.log(aboutMe(people))
```

Comment:
Thi sgets the job done but there are tools that would allow this process to be more dynamic and allow for use cases that are not a part of these testing variables. ALWAYS consider the edge cases.